nimble/host/src/ble_sm.c

```diff
@@ -933,6 +933,12 @@ ble_sm_process_result(uint16_t conn_handle, struct ble_sm_result *res)
 
         ble_hs_unlock();
 
+        if (res->enc_cb &&
+            res->app_status != BLE_HS_ENOTCONN) {
+            /* Do not send this event on broken connection */
+            ble_gap_pairing_complete_event(conn_handle, res->sm_err);
```

Comment:
shouldn't we send here:

res->app_status = BLE_HS_SM_PEER_ERR(cmd->reason); ?

Reply:
I think we should send in event `res->sm_err` which is either BLE_SM_ERR_SUCCESS (if set by `ble_sm_key_exch_success`) or error code (if set by `ble_sm_key_exch_success`), but maybe I don't see something?